tests/plotting/test_plotting.py

```diff
@@ -206,6 +208,16 @@ def __init__(
         self.var_error_value = var_error_value
         self.var_warning_value = var_warning_value
 
+        if high_variance_tests is not None:
+            self.high_variance_tester = re.compile(r"|".join(high_variance_tests))
```

Comment:
Thanks for the ping :)

For your first question, the `r` is unnecessary. It's not a regex thing _per se_, rather a way to change how stuff in the string literal get escaped. Consequently once the string is evaluated the "r" evaporates. Here's what I mean:
```py
>>> r'|'
'|'
```
The literal `r'|'` evaluates to `'|'`. This is why I prefer to call these raw string _literals_: they change how the literal is evaluated. So the rawness only matters when you have potential escape characters in there, i.e. backslashes:
```py
>>> r'\n'
'\\n'
>>> '\n'
'\n'
```
The former (raw) string literal auto-escapes the backslash and gives you a 2-length string literal, where the first character is a literal backslash. The latter (regular? well-done?) string literal applies the escape sequence giving you a 1-length string literal with the linefeed character inside.

So by the time any regex gets into the picture the raw string literal is converted to a regular old string with some additional backslashes. The reason why this is often coupled to regex is that for regex you often want to escape certain characters, so you want to have a literal backslash followed by a `.` to mean "literal `.`" in regex. So you often end up wanting to have backslash literals in your string, for which using a raw string literal is very handy. But also, as you saw, you can't have the backslash in the last position because that would still escape the closing quote.

As for your other question, I'm not sure about performance considerations. As long as we just have literal substrings concatenated with `'|'`, lookup is probably as fast (or even faster if searching happens in C land) as an equivalent Python loop would be. And I don't know how long compilation takes; we could time this, but I doubt it's significant in case of a handful of strings. If we're worried for performance we could also support compiled patterns rather than iterables of strings, in which case users could choose to pre-compile their patterns (if this makes sense).

Reply:
Think I got it.  Thanks!  I will remove the raw string here as it is unnecessary.

I like the idea that if we run into performance benchmarks in the future, we could allow users to pass in precompiled regex.  This seems like the best of both worlds answer.  I'll leave it out if this implementation as I suspect it might never be needed.  I mainly wanted to avoid needing a large breaking restructure in the future.